lib/ansible/playbook/conditional.py

```diff
@@ -113,21 +114,16 @@ def _check_conditional(self, conditional, templar, all_vars):
         if isinstance(conditional, bool):
             return conditional
 
+        if C.CONDITINAL_BARE_VARS:
+            if conditional in all_vars and VALID_VAR_REGEX.match(conditional):
+                display.deprecated('evaluating %s as a bare variable, this behaviour will go away and you might need to add |bool'
```

Comment:
Would be clearer to say we are automatically converting to bool.  "evaluating as a bare variable" didn't make sense to me and makes it sound like `when: enable_my_feature` is improper usage.  (My roles require the caller to pass bool's as such, not as strings.)

Reply:
but we are not automatically converting to a bool in this case, we are using the variable in 'python bool context' .. which might make it behave as if converting,  vs 'jinja2 if expression' which has slightly diff rules.